test/samples/required/expected.json

```diff
@@ -25,9 +25,33 @@
 						"type": "string"
 					}
 				],
-				"name": "dates_record",
+				"name": "dates_required_record",
 				"type": "record"
 			}
+		},
+		{
+			"doc": "",
+			"name": "dates_optional",
```

Comment:
Can this be a new sample has having an optional check in a required test is not very readable.